node/aas-aid/src/asset-interfaces-description.ts

```diff
@@ -1498,6 +1524,56 @@ export class AssetInterfacesDescription {
                         });
                     }
 
+                    // Handling of any key that has : in it by creating semantic ids for each
+                    // no need to bother if there is no object in the context or it is a string
+                    const tdContext = td["@context"];
+                    let prefixObject: any = {};
+
+                    if (Array.isArray(tdContext)) {
+                        if (tdContext.length > 3) {
+                            // if it is longer than 3, assume 3rd item is the object. This offset is needed as node-wot always adds @language at the end
```

Comment:
This offset can be avoided if we do not normalize the TD when parsing. In any case, I find it a bit weird that node-wot changes the context.

Also this really assumes that we have the TD contexts (v1 or v1.1) and then an object with prefixes.

Reply:
Mhh, not sure what is meant by "node-wot changes the context". Should we open an issue on node-wot side?
Anyhow, it should work in any order ... but I do recall that in the TD spec there is a order defined... maybe that is the reason?

![image](https://github.com/user-attachments/assets/72e54d57-06bd-4f09-9c57-f00820a0a1b5)

Reply:
In general I think this part of the code assumes some characteristics that might change over time in node-wot. Not sure if this future proof...

Reply:
I also feel that this is not future-proof but I have thought of a small improvement: Going through all object items in the array and merging them into one object.

node-wot is not doing anything against the TD spec but it always adds an object with `@language` in it. So the TD I provide is changed after the parsing.

Reply:
This is addressed now